apps/mobile/src/slices/auth/actions.ts

```diff
@@ -1,21 +1,37 @@
 import { createAsyncThunk } from "@reduxjs/toolkit";
 
+import { storage, StorageKey } from "~/libs/packages/storage/storage";
 import { type AsyncThunkConfig } from "~/libs/types/types";
 import {
+	type UserDto,
 	type UserSignUpRequestDto,
 	type UserSignUpResponseDto,
 } from "~/packages/users/users";
 
 import { name as sliceName } from "./auth.slice";
 
+const getAuthenticatedUser = createAsyncThunk<
+	UserDto,
+	undefined,
+	AsyncThunkConfig
+>(`${sliceName}/get-authenticated-user`, async (_, { extra }) => {
+	const { authApi } = extra;
+	const response: UserDto = await authApi.getAuthenticatedUser();
+
+	return response;
+});
+
 const signUp = createAsyncThunk<
-	UserSignUpResponseDto,
+	UserDto,
 	UserSignUpRequestDto,
 	AsyncThunkConfig
->(`${sliceName}/sign-up`, (signUpPayload, { extra }) => {
+>(`${sliceName}/sign-up`, async (signUpPayload, { extra }) => {
 	const { authApi } = extra;
 
-	return authApi.signUp(signUpPayload);
+	const response: UserSignUpResponseDto = await authApi.signUp(signUpPayload);
```

Comment:
```suggestion
	const {user, token} = await authApi.signUp(signUpPayload);
```
Same here, do we need it?
Also use desctructuring

Reply:
Update